packages/snaps-controllers/src/types/encryptor.ts

```diff
@@ -0,0 +1,117 @@
+import type { Json } from '@metamask/utils';
+
+export type PBKDF2Params = {
+  iterations: number;
+};
+
+export type KeyDerivationOptions = {
+  algorithm: 'PBKDF2';
+  params: PBKDF2Params;
+};
+
+export type EncryptionResult = {
+  data: string;
+  iv: string;
+  salt: string;
+  keyMetadata?: KeyDerivationOptions;
+};
+
+/**
+ * A generic encryptor interface that supports encrypting and decrypting
+ * serializable data with a password.
+ */
+export type GenericEncryptor = {
+  /**
+   * Encrypts the given object with the given password.
+   *
+   * @param password - The password to encrypt with.
+   * @param object - The object to encrypt.
+   * @returns The encrypted string.
+   */
+  encrypt: (password: string, object: Json) => Promise<string>;
+  /**
+   * Decrypts the given encrypted string with the given password.
+   *
+   * @param password - The password to decrypt with.
+   * @param encryptedString - The encrypted string to decrypt.
+   * @returns The decrypted object.
+   */
+  decrypt: (password: string, encryptedString: string) => Promise<unknown>;
+  /**
+   * Optional vault migration helper. Checks if the provided vault is up to date
+   * with the desired encryption algorithm.
+   *
+   * @param vault - The encrypted string to check.
+   * @param targetDerivationParams - The desired target derivation params.
+   * @returns The updated encrypted string.
+   */
+  isVaultUpdated: (
+    vault: string,
+    targetDerivationParams?: KeyDerivationOptions,
+  ) => boolean;
```

Comment:
```suggestion
  encrypt: (password: string, object: Json) => Promise<string>;

  /**
   * Decrypts the given encrypted string with the given password.
   *
   * @param password - The password to decrypt with.
   * @param encryptedString - The encrypted string to decrypt.
   * @returns The decrypted object.
   */
  decrypt: (password: string, encryptedString: string) => Promise<unknown>;

  /**
   * Optional vault migration helper. Checks if the provided vault is up to date
   * with the desired encryption algorithm.
   *
   * @param vault - The encrypted string to check.
   * @param targetDerivationParams - The desired target derivation params.
   * @returns The updated encrypted string.
   */
  isVaultUpdated: (
    vault: string,
    targetDerivationParams?: KeyDerivationOptions,
  ) => boolean;
```

Reply:
It's not obvious to me what this suggestion is trying to do

Reply:
New line between function definitions. I did it as separate suggestions in other review comments.